brontide/noise.go

```diff
@@ -739,6 +740,15 @@ func (b *Machine) ReadMessage(r io.Reader) ([]byte, error) {
 		return nil, err
 	}
 
+	// If this is the first message being read, take a read buffer from the
+	// buffer pool. This is delayed until this point to avoid allocating
+	// read buffers until after the peer has successfully completed the
+	// handshake, and is ready to begin sending lwnire messages.
+	if b.nextCipherText == nil {
+		b.nextCipherText = readBufferPool.Take()
+		runtime.SetFinalizer(b, freeReadBuffer)
```

Comment:
👍

Reply:
Have we confirmed that this is indeed being executed in a timely manner, and isn't left dangling due to a pointer in some struct?

Reply:
when running on my node the profiles didn't hint at this happening. it matched what was being done for the write pools w/o the finalizer